pxtcompiler/emitter/service.ts

```diff
@@ -1306,13 +1312,19 @@ namespace ts.pxtc.service {
                     host.opts.skipPxtModulesEmit = true
             }
             let ts2asm = compile(host.opts, service)
-            res = { sourceMap: res.sourceMap, ...ts2asm }
-            if (res.needsFullRecompile) {
-                pxt.log("trigering full recompile")
+            res = {
+                sourceMap: res.sourceMap,
+                ...ts2asm,
+            }
+            if (res.needsFullRecompile || ((!res.success || res.diagnostics.length) && host.opts.clearIncrBuildAndRetryOnError)) {
```

Comment:
was running into some occasional (~5 times in arcade checkdocs) errors where having it retry here fixed things. This seems to imply a bug in the incremental compiles to me, but having the param to do a full recompile on error seemed fine to me.

Reply:
@mmoskal fyi